[shelly] NPE during discovery

On 5.0.0.M4, I found the following in the log when starting my instance:

```
09:01:19.406 [ERROR] [ry.mdns.internal.MDNSDiscoveryService] - Participant 'org.openhab.binding.shelly.internal.discovery.ShellyMDNSDiscoveryParticipant' threw an exception
java.lang.NullPointerException: Cannot invoke "String.isEmpty()" because "deviceType" is null
        at org.openhab.binding.shelly.internal.discovery.ShellyThingCreator.getThingTypeUID(ShellyThingCreator.java:544) ~[?:?]
        at org.openhab.binding.shelly.internal.discovery.ShellyBasicDiscoveryService.createResult(ShellyBasicDiscoveryService.java:145) ~[?:?]
        at org.openhab.binding.shelly.internal.discovery.ShellyMDNSDiscoveryParticipant.createResult(ShellyMDNSDiscoveryParticipant.java:136) ~[?:?]
        at org.openhab.core.config.discovery.mdns.internal.MDNSDiscoveryService.createDiscoveryResult(MDNSDiscoveryService.java:222) ~[?:?]
        at org.openhab.core.config.discovery.mdns.internal.MDNSDiscoveryService.considerService(MDNSDiscoveryService.java:214) ~[?:?]
        at org.openhab.core.config.discovery.mdns.internal.MDNSDiscoveryService.serviceResolved(MDNSDiscoveryService.java:207) ~[?:?]
        at javax.jmdns.impl.ListenerStatus$ServiceListenerStatus.serviceResolved(ListenerStatus.java:106) ~[?:?]
        at javax.jmdns.impl.JmDNSImpl$1.run(JmDNSImpl.java:911) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:317) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
        at java.lang.Thread.run(Thread.java:1583) [?:?]
```

Clearly, this should not happen.

--- a/bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/discovery/ShellyBasicDiscoveryService.java
+++ b/bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/discovery/ShellyBasicDiscoveryService.java
@@ -13,6 +13,7 @@
 package org.openhab.binding.shelly.internal.discovery;
 
 import static org.openhab.binding.shelly.internal.ShellyBindingConstants.*;
+import static org.openhab.binding.shelly.internal.discovery.ShellyThingCreator.*;
 import static org.openhab.binding.shelly.internal.util.ShellyUtils.*;
 import static org.openhab.core.thing.Thing.*;
 
@@ -131,13 +132,10 @@ public class ShellyBasicDiscoveryService extends AbstractDiscoveryService {
             api.initialize();
             devInfo = api.getDeviceInfo();
             mac = getString(devInfo.mac);
-            model = devInfo.type;
+            model = getString(devInfo.type);
             auth = getBool(devInfo.auth);
-            if (name.isEmpty() || name.startsWith("shellyplusrange")) {
+            if (name.isEmpty() || name.startsWith(SERVICE_NAME_SHELLYPLUSRANGE_PREFIX)) {
                 name = devInfo.hostname;
-            }
-            if (devInfo.name != null) {
-                deviceName = getString(devInfo.name);
             }
 
             thingType = substringBeforeLast(name, "-");

--- a/bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/discovery/ShellyThingCreator.java
+++ b/bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/discovery/ShellyThingCreator.java
@@ -151,6 +151,7 @@ public class ShellyThingCreator {
 
     // Service name prefixes
     public static final String SERVICE_NAME_SHELLYBLU_PREFIX = "shellyblu";
+    public static final String SERVICE_NAME_SHELLYPLUSRANGE_PREFIX = "shellyplusrange";
 
     private static final String SERVICE_NAME_SHELLY2_PREFIX = "shellyswitch";
     private static final String SERVICE_NAME_SHELLY25_PREFIX = "shellyswitch25";

--- a/bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java
+++ b/bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java
@@ -610,7 +610,7 @@ public abstract class ShellyBaseHandler extends BaseThingHandler
                 && prf.status.rangeExtender.apClients != null) {
             for (Shelly2APClient client : profile.status.rangeExtender.apClients) {
                 String secondaryIp = config.deviceIp + ":" + client.mport.toString();
-                String name = "shellyplusrange-" + client.mac.replaceAll(":", "");
+                String name = SERVICE_NAME_SHELLYPLUSRANGE_PREFIX + "-" + client.mac.replaceAll(":", "");
                 DiscoveryResult result = ShellyBasicDiscoveryService.createResult(true, name, secondaryIp,
                         bindingConfig, httpClient, messages);
                 if (result != null) {

